tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/listconstructor/ListConstructorExprTest.java

```diff
@@ -61,10 +61,12 @@ public void diagnosticsTest() {
                 "inferred for '[1, p]'", 38, 35);
         BAssertUtil.validateError(resultNegative, i++, "invalid list constructor expression: types cannot be " +
                 "inferred for '[a, 4]'", 41, 23);
-        BAssertUtil.validateError(resultNegative, i++, "tuple and expression size does not match",
+        BAssertUtil.validateError(resultNegative, i++,
+                "invalid usage of list constructor: type 'NoFillerObject' does not have a filler value",
                 45, 31);
-        BAssertUtil.validateError(resultNegative, i++, "tuple and expression size does not match",
-                46, 56);
+        BAssertUtil.validateError(resultNegative, i++,
+                "invalid usage of list constructor: type '[NoFillerObject,NoFillerObject]'" +
```

Comment:
If the object has no parameters in init method and no error type, the `new T()` should be the filler value right?

Reply:
it works for classes

Reply:
Seems this is a bug in SL as well. Added a comment in #35712